### ISSUE TITLE
[ON HOLD] [MEDIUM] [Splits] [$250] Split bill - Split amount is lost when adding receipt and splitting bill

### ISSUE THREAD

**Issue by izarutskaya** [Expensify/App#37491]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** v1.4.45-0
**Reproducible in staging?:** Y
**Reproducible in production?:** N
**Email or phone of affected tester (no customers):** natnael.expensify+3@gmail.com
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Issue reported by:** Applause-Internal Team

## Action Performed:
1. Click on 'FAB' > Request money > enter amount
2. Select a user > Add receipt > return to where you can select a user
3. Click on 'Split' button
4. Click on 'Add to split' > click on 'Split'

## Expected Result:
Entered amount shouldn't be lost, and request shouldn't change to scan request

## Actual Result:
Entered amount is lost and request is changed to a scan request

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [x] Android: Native
- [x] Android: mWeb Chrome
- [ ] iOS: Native
- [x] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [x] MacOS: Desktop

## Screenshots/Videos

https://github.com/Expensify/App/assets/115492554/316ac613-a1bf-4035-972d-c2ae29b23716

</details>

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01e554328af9becb46</li>
        <li>Upwork Job ID: 1763525709189742592</li>
        <li>Last Price Increase: 2024-08-08</li>
<li>Automatic offers: </li>
<ul>
<li>shubham1206agra | Reviewer | 0</li>
<li>rojiphil | Contributor | 0</li>
</ul></li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @youssef-lr</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @youssef-lr</details>

**Comment by izarutskaya** [comment]
@grgia I haven't added the External label as I wasn't 100% sure about this issue. Please take a look and add the label if you agree it's a bug and can be handled by external contributors.

**Comment by grgia** [comment]
I believe this is happening with manual requests too

**Comment by rojiphil** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.

When `Split bill` is used with a receipt, the split amount is lost in the split bill details page.

### What is the root cause of that problem?

[Here](https://github.com/Expensify/App/blob/f4089c096864072ee3dfd1f0aee260cc2a8d3268/src/pages/iou/request/step/IOURequestStepConfirmation.js#L262), when we split the bill with a receipt, it was thought to be done via the scan tab where we do not enter the amount. In such a case, the flow starts with a split bill request and is completed later. However, as done in this issue, it is also possible to split the bill via `Manual` request with an added `receipt`. Currently, this goes through the default `startSplit` flow. This is the root cause.  

I do not think this is a regression. 

### What changes do you think we should make in order to solve the problem?

We can support split bill request for manual request type with receipt by making use of the `splitBill` [here](https://github.com/Expensify/App/blob/f4089c096864072ee3dfd1f0aee260cc2a8d3268/src/pages/iou/request/step/IOURequestStepConfirmation.js#L280) and `splitBillAndOpenReport` [here](https://github.com/Expensify/App/blob/f4089c096864072ee3dfd1f0aee260cc2a8d3268/src/pages/iou/request/step/IOURequestStepConfirmation.js#L299). 

1) First, we need to avoid going through the `startSplitBill` flow by additionally checking for a valid amount like this [here](https://github.com/Expensify/App/blob/f4089c096864072ee3dfd1f0aee260cc2a8d3268/src/pages/iou/request/step/IOURequestStepConfirmation.js#L262)
 
            if (iouType === CONST.IOU.TYPE.SPLIT && receiptFile && transaction.amount === 0) {

2) Now, the flow with either go through `splitBill` [here](https://github.com/Expensify/App/blob/f4089c096864072ee3dfd1f0aee260cc2a8d3268/src/pages/iou/request/step/IOURequestStepConfirmation.js#L280) or `splitBillAndOpenReport` [here](https://github.com/Expensify/App/blob/f4089c096864072ee3dfd1f0aee260cc2a8d3268/src/pages/iou/request/step/IOURequestStepConfirmation.js#L299). Here, let us pass the `receiptFile` into `splitBill` and `splitBillAndOpenReport` as it is done [here](https://github.com/Expensify/App/blob/f4089c096864072ee3dfd1f0aee260cc2a8d3268/src/pages/iou/request/step/IOURequestStepConfirmation.js#L270). 

3) Inside the `createSplitsAndOnyxData` [here](https://github.com/Expensify/App/blob/f4089c096864072ee3dfd1f0aee260cc2a8d3268/src/libs/actions/IOU.ts#L1621), generate the `receiptObject` like it is done [here](https://github.com/Expensify/App/blob/f4089c096864072ee3dfd1f0aee260cc2a8d3268/src/libs/actions/IOU.ts#L2091C1-L2092C52) and add this to the transaction [here](https://github.com/Expensify/App/blob/f4089c096864072ee3dfd1f0aee260cc2a8d3268/src/libs/actions/IOU.ts#L1631C1-L1632C19) and iou report action [here](https://github.com/Expensify/App/blob/f4089c096864072ee3dfd1f0aee260cc2a8d3268/src/libs/actions/IOU.ts#L1652). Also, failure data can consider receipt error using `getReceiptError` like [here](https://github.com/Expensify/App/blob/f4089c096864072ee3dfd1f0aee260cc2a8d3268/src/libs/actions/IOU.ts#L2234)

However, this would also need BE support as `SplitBill` API need to send back `receipt` in response.

### What alternative solutions did you explore? (Optional)

Alternatively, if we want to do this via `startSplitBill` flow, we can include `amount` in the flow. Here, I think even BE needs to ensure that transaction contains the `amount`. Happy to explore more on this if we want to go through this approach.

**Comment by puneetlath** [comment]
FYI, this is likely fixed by [this revert](https://github.com/Expensify/App/pull/37529). Will have QA re-test once it is CP'd.

**Comment by puneetlath** [comment]
Looks like that revert ^ didn't fix the issue. So we still need to figure out what is going on here @grgia.

**Comment by situchan** [comment]
Another regression from https://github.com/Expensify/App/pull/33482
Not able to reproduce after reverting that PR locally

**Comment by youssef-lr** [comment]
@situchan if you revert that PR you will not be able to split with a single user, which is why you can't reproduce this. I think this is not a regression as @rojiphil mentioned. Confirming shortly.

**Comment by situchan** [comment]
ok, if this bug happens on production when split bill with workspace (not single user), not blocker.

**Comment by youssef-lr** [comment]
>if this bug happens on production when split bill with workspace (not single user)

have you tested this?

**Comment by situchan** [comment]
checking

**Comment by youssef-lr** [comment]
we can't test that scenario in prod, because the PR that fixes detecting 'Split' from the route is only in staging.

https://github.com/Expensify/App/assets/9680864/6affccc1-f255-4aab-a167-e18e7a5a8f8a

**Comment by youssef-lr** [comment]
@bernhardoj I think this is an edge case we missed, do you mind taking a look?

**Comment by situchan** [comment]
> we can't test that scenario in prod, because the PR that fixes detecting 'Split' from the route is only in staging. Notice how the request never becomes a split.

👍

**Comment by youssef-lr** [comment]
I think demoting is fine here.

**Comment by bernhardoj** [comment]
Checking

**Comment by rojiphil** [comment]
As the OP mentions, I think the issue is that the request changes from a `manual` split with receipt to a `scan` split. This results in a `startSplitBill` flow which does not seem right. And this problem can also be reproduced in production as shown in the test video below. Further, the code referencing this issue has not changed for quite some time as is evident in the [proposal](https://github.com/Expensify/App/issues/37491#issuecomment-1971420826). So, I still think this is not a regression. 
What do you think?
cc @grgia @youssef-lr @bernhardoj @situchan 


https://github.com/Expensify/App/assets/3004659/8e1efba9-ec16-4d89-8ba9-e1f05d2ada85

**Comment by bernhardoj** [comment]
Ok, I agree with @rojiphil too that this is not a regression. The PR has been deployed to prod, so I tested on an older branch and can reproduce this when splitting with multiple users.

https://github.com/Expensify/App/assets/50919443/1dea60dd-8fe1-4e32-9fee-81e42169866e

**Comment by situchan** [comment]
Agree this can be treated as separate issue, not regression

**Comment by youssef-lr** [comment]
@grgia grabbing this one if you don't mind!

**Comment by rojiphil** [comment]
> However, this would also need BE support as `SplitBill` API need to send back `receipt` in response.

@youssef-lr Thanks for assigning me here. 
This issue also needs BE support as mentioned above
Once the BE is ready I will raise the PR in FE.

@shubham1206agra Welcome to the party 🎉  here.

**Comment by rojiphil** [comment]
@youssef-lr  Another issue [here](https://github.com/Expensify/App/issues/35961#issuecomment-1931809353) related to distance requests also needs BE support so that `SplitBill` API sends back `receipt` for distance requests. Since this is also related, not sure if you want to pick that up too for BE changes.

**Comment by youssef-lr** [comment]
Thanks for bringing that up @rojiphil, I will take a look on Monday.

**Comment by youssef-lr** [comment]
Still planning to get to this today.

**Comment by youssef-lr** [comment]
@rojiphil this will indeed to be held on backend changes, the manual split bill backend should support adding receipts, currently only `startSplitBill` supports it.

**Comment by youssef-lr** [comment]
I still haven't had time for this yet. Will work on the backend PR next week.

**Comment by isabelastisser** [comment]
Hi @youssef-lr, any updates?

**Comment by youssef-lr** [comment]
I haven't had time to get to this yet, I was OOO sick when you mentioned me @isabelastisser, last week I have been prioritizing a wave issue and a vip-split design doc

**Comment by isabelastisser** [comment]
Hey @youssef-lr, any updates? Thanks!

**Comment by youssef-lr** [comment]
I haven't been able to switch focus to this one. Uneven splits is almost done, I'll be able to get back to this this week.

**Comment by isabelastisser** [comment]
Any progress @youssef-lr ? Thanks!

**Comment by youssef-lr** [comment]
No progress yet sadly, I haven't been able to get to this yet as I'm focusing on higher priority issues.

**Comment by youssef-lr** [comment]
@JmillsExpensify this issue is about allowing to add a receipt to manual splits, I remember you mentioned we'll have a new pattern for this, do you think we should wait for that?

**Comment by isabelastisser** [comment]
Bump @JmillsExpensify

**Comment by JmillsExpensify** [comment]
@youssef-lr Blast from the past. At one point this was going to be an option row in the `Details` page for the report, though we never ended up adding it. Further, we kept the add receipt option in the three dot overflow menu. Then additionally, we ended up deciding to show an empty receipt state for all `expense` transactions. 

@Expensify/design I'm curious for your take on this one? Should add an `Add receipt` option in the details page for manual splits? I'm not super convinced we should go the receipt empty state route for consumers, as that feels like it's promoting receipts to best practice when they are completely optional.

**Comment by dannymcclain** [comment]
Honestly I don't know that I feel super duper strongly. I could definitely see `Receipt` just being a push input like Description, Merchant, etc. So we're not really super calling attention to it (like the empty state does) but we're still giving people the option.

**Comment by dubielzyk-expensify** [comment]
Yeah the overflow isn't ideal here is it 😅 I don't haaate the UI though for if they manage to add it, but for the empty state if would definitely take precedence. Might just be me, but I'd love to find a consistent version that works for both split and manual even if it means that the receipt empty state on manual gets scaled down quite a bit. Is that too spicy?

**Comment by JmillsExpensify** [comment]
Haha not necessarily. I'm not too passionate either. Split is still pretty aspirational, so I could potentially get down with just leaning into the empty state for now.

**Comment by shawnborton** [comment]
I like the idea of the push row for Receipt to de-emphasize it quite a bit.

**Comment by dannymcclain** [comment]
>I like the idea of the push row for Receipt to de-emphasize it quite a bit.

It's definitely not the first time it's come up, and I do feel like it could easily work for both split and manual without being too in the way. 🤷‍♂️

**Comment by JmillsExpensify** [comment]
Let's just do it then? So we'd use this pattern on splits/IOUs and keep the empty state for expense reports, right?

**Comment by shawnborton** [comment]
I can get down with that, yeah!

**Comment by dannymcclain** [comment]
Yeah that's what I was thinking too. Something like this?

![image](https://github.com/Expensify/App/assets/7516624/68c0c028-47ce-4370-8864-8d09d484d3a3)

Is this the right placement for the `Receipt` field?

**Comment by dannymcclain** [comment]
Yeah that's fair, I just wasn't sure. So if we put it in the same place, something like this?

![image](https://github.com/Expensify/App/assets/7516624/cffa4977-b87b-4afe-9fab-740a8318702b)

And here's the submit expense flow with receipt empty state for reference:
![image](https://github.com/Expensify/App/assets/7516624/729948cc-c3da-416b-be12-b00fe8f1f07e)

**Comment by JmillsExpensify** [comment]
My preference, to keep this simple, would be to not show the receipt in the creation flow. Instead, we just add an option row to the details page post-creation. Basically, I feel like putting the receipt more front-end center in creation flow feels like a bigger change, and we still expect most split/iou users to never need a receipt in the manual flow.

**Comment by dannymcclain** [comment]
Ahhh ok I was misunderstanding that.

**Comment by shawnborton** [comment]
Or we could tuck it under "Show more" for those flows?

**Comment by JmillsExpensify** [comment]
We could do that too, though if you end up adding a receipt, then I think we need to show the receipt in the same place as expenses.

**Comment by dubielzyk-expensify** [comment]
I don't mind where it's just a push row, but in the "Show more" works too

**Comment by JmillsExpensify** [comment]
Alright, so it sounds like we're largely decided here? Can we get an updated set of mocks for clarity?

**Comment by dannymcclain** [comment]
>Alright, so it sounds like we're largely decided here? Can we get an updated set of mocks for clarity?

I'm sorry, can you summarize what we decided? I actually can't figure it out from the thread haha.

**Comment by JmillsExpensify** [comment]
I think we decided:
 - We will show the ability to add a receipt in the creation flow for IOU/split. It'll be a push row under `Show more`, since receipts are never required for IOU/Split.
 - If added, the receipt will always appear in the same position. So above the detailed expense data like description, etc. 
 - Post creation, if a receipt isn't added, then it'll similarly be a push row in the Details page. 
 
 Is that right, or did I misunderstand?

**Comment by shawnborton** [comment]
For IOUs, we always put the receipt under the `To:` field. So for split, does it make sense to put it under all of the people rows, like so?
![CleanShot 2024-06-11 at 09 50 43@2x](https://github.com/Expensify/App/assets/2319350/fe6bf1e9-1f10-49eb-9c05-a1413a9e2b0b)

(I started exploring this in this [Figma file here](https://www.figma.com/design/i69DCNjwiWaux3GSClfXLx/%23vip-split%3A-Split-expense-in-Global-Create?node-id=693-21921&t=3urqfYU6H58Y6kaO-1) btw, wasn't sure where to put it!)

**Comment by dubielzyk-expensify** [comment]
I like that 👍

**Comment by dannymcclain** [comment]
What I don't really love is how the filled receipt is in a different spot than the empty push row. I find it to be pretty weird that the order of the information changes based on whether things are filled out or not.

Also, wouldn't we label that push row `Receipt` instead of `Add receipt`? And then show the little `Receipt` label above the image?

**Comment by shawnborton** [comment]
Good point that it should just say "Receipt" and that we'd maintain a small label above the thumbnail. Looks like you beat me to that but those updates to my mocks look like this:
![CleanShot 2024-06-11 at 17 17 01@2x](https://github.com/Expensify/App/assets/2319350/9e496678-6acc-4a01-a486-e9c1bdf40cd6)

I think that's a fair point about the receipt position though, curious what @JmillsExpensify thinks as he mentioned this:

>If added, the receipt will always appear in the same position. So above the detailed expense data like description, etc.

**Comment by isabelastisser** [comment]
Not overdue, it looks like we're still discussing this @shawnborton @dannymcclain

**Comment by JmillsExpensify** [comment]
Yes, I still think it's weird that we're switching the receipt position around. I do think it's a fair point that the read-only split details/preview is different than than the expense details/preview, but I still find is weird that we're changing the location of the receipt around between types.

**Comment by shawnborton** [comment]
Hmm so then I think the options are:
- move the "Receipt" push row to always be in the same spot everywhere, which sounds like it would be directly under the To field
- keep the "Add receipt" option in the overflow menu in the top right of the page header, and when the receipt is added, we always show the receipt in the same spot

Does that sound right?

**Comment by dannymcclain** [comment]
Yeah that sounds right to me Shawn. Personally I'm in favor of your first option because I don't think the push rows are super in the way. If you don't want to add a receipt, just skip over the input. No big deal. IMO, a single empty push input is not so obtrusive that we desperately need to hide it.

**Comment by shawnborton** [comment]
Okay so that would essentially be your mockup you made here:
![image](https://github.com/Expensify/App/assets/2319350/c2714007-fefe-45c8-aa26-b5c6426d8f45)


Which personally I don't mind. But as Jason says:

> Basically, I feel like putting the receipt more front-end center in creation flow feels like a bigger change, and we still expect most split/iou users to never need a receipt in the manual flow.

So then maybe we just do nothing in terms of where we place the empty state or the option to add the receipt, since it already exists in the overflow menu for non-workspace flows:
![CleanShot 2024-06-19 at 16 10 29@2x](https://github.com/Expensify/App/assets/2319350/fde42ad7-9faf-437f-9ce7-7fc2ed0884bd)

So back to the original question:

> Should add an Add receipt option in the details page for manual splits?

I would think we'd just reuse the overflow menu for it then.

**Comment by dannymcclain** [comment]
>I would think we'd just reuse the overflow menu for it then.

I'm totally fine with this approach. Basically do nothing for IOUs and make Splits work the same way, right?

Though because I can't help myself 😅, my 2 cents is that adding an empty push input is not really making adding a receipt front and center. But I don't feel SO strongly as to keep pushing for it if we're happy to just copy how IOUs currently work for this.

**Comment by shawnborton** [comment]
> Basically do nothing for IOUs and make Splits work the same way, right?

Exactly. Let's see what Jason thinks though :)

**Comment by JmillsExpensify** [comment]
Haha I'm also in agreement. Let's bring splits in line with IOU/expense and add the option to the overflow menu.

**Comment by dannymcclain** [comment]
Sounds good. Once they're the same, we can always decide later to update both to something different if we want. But yeah, just getting them in line will be a great first step.

**Comment by JmillsExpensify** [comment]
What are the next steps on this PR?

**Comment by youssef-lr** [comment]
@rojiphil are you able to work on this soon?

**Comment by rojiphil** [comment]
> are you able to work on this soon?

Lot to unpack here but, yes, will pick this up today in some time.

**Comment by rojiphil** [comment]
> can you please provide an update? Thanks!

The current status is that @youssef-lr is taking a look (mentioned [here](https://github.com/Expensify/App/pull/44496#issuecomment-2221302024)) at any needed BE changes for this feature based on the FE changes made thus far in a draft [PR](https://github.com/Expensify/App/pull/44496)

**Comment by youssef-lr** [comment]
I haven't had time to prioritize this yet, I'm on it next week.

**Comment by isabelastisser** [comment]
Not overdue. Any updates, @youssef-lr? Thanks!

**Comment by youssef-lr** [comment]
No update yet @isabelastisser :/, I'm focusing on Aug 15th issues

**Comment by isabelastisser** [comment]
Any updates here @youssef-lr ? thanks!

**Comment by youssef-lr** [comment]
I think we can put this on monthly for as long as vip-split is paused

**Comment by isabelastisser** [comment]
vip-split is paused; issue is on hold.

**Comment by isabelastisser** [comment]
Not overdue.

**Comment by youssef-lr** [comment]
On hold.

**Comment by isabelastisser** [comment]
not overdue, still on hold.

**Comment by isabelastisser** [comment]
@youssef-lr, should we still hold this? What are the pending tasks/requirements before we lift the hold and continue working on it? thanks!

**Comment by isabelastisser** [comment]
@youssef-lr, can you please see my message above and provide an update? Thanks!

**Comment by youssef-lr** [comment]
@isabelastisser yes, vip-split is still paused.

**Comment by isabelastisser** [comment]
Still on hold.

**Comment by isabelastisser** [comment]
Not overdue.

**Comment by isabelastisser** [comment]
Still on hold.

**Comment by isabelastisser** [comment]
Still on hold.

**Comment by isabelastisser** [comment]
@youssef-lr, should this be closed?

**Comment by youssef-lr** [comment]
I think we can just close this as we're not prioritizing IOU splits currently.

**Comment by mallenexpensify** [comment]
<img width="322" height="152" alt="Image" src="https://github.com/user-attachments/assets/12ece81d-a8c4-4a4f-b73f-3688081ba15f" />

woah... that's a lotta comments.  

If a contributor was hired and  PR created that was reviewed by the C+, both are due 100%.  
@rojiphil , @shubham1206agra can you confirm the above? If so, @kadiealexander , can you plz pay? Thx

**Comment by kadiealexander** [comment]
### Payment Summary

- [ ] Contributor: @rojiphil due $250 via NewDot
- [ ] Contributor+: @shubham1206agra due $250 via NewDot

**Comment by garrettmknight** [comment]
$250 approved for @shubham1206agra
